### PR TITLE
Fix error with zero-length Arrow response bodies.

### PR DESCRIPTION
**Pull request by thetorpedodog**
For situations where a UDF was successful but for whatever reason
the result returned was a zero-length body, we would mistakenly assume
that the body was not provided, rather than recognizing it as a valid
but zero-length body. We now distinguish between no body and a
zero-length body, and properly handle zero-length Arrow bodies.

We had previously tested against empty Arrow bodies with SQL queries.
SQL queries include a schema in their results, and thus have nonzero
response length. However, empty UDF responses which are written in Arrow
format would not have any write calls at all, meaning the response was
completely empty.

Also adds a test for this situation.

---

Thanks to @ckrilow for finding this bug!

### CODE REVIEW

File: tiledb/cloud/_results/results.py [hunk 1]
```diff
@@ -101,7 +101,7 @@ def _validate(self, attribute, value):
         del attribute, value  # unused
         if self.results_stored and not self.task_id:
             raise ValueError("task_id must be set for stored results")
-        if not self.results_stored and not self._body:
+        if not self.results_stored and self._body is None:
```
Comment (johnkerl) [inline tiledb/cloud/_results/results.py:104]: Interesting this is `None` vs `""` -- I do believe you've tested this correctly. Is it worth adding

```
if not self.results_stored and (self._body is None or self._body == ""):
```

?

(just a thought)

**Review by johnkerl — APPROVED** [review]
LGTM, with comment

File: tiledb/cloud/_results/results.py (hunk 1 above)
Comment (thetorpedodog) [reply tiledb/cloud/_results/results.py:104]: We specifically don’t; this needs to test that the `_body` is not provided (results were not downloaded) versus empty (results were downloaded and were zero-length).
